### PR TITLE
MM-48838 : Introduce formatjs eslint plugin and add multiple spaces rules

### PR DESCRIPTION
**Pull request by M-ZubairAhmed**
#### Summary
- Official `eslint-plugin-formatjs` plugin for formatjs added.
- Added the rule to not allow multiple white spaces from default messages.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48838

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note

```release-note
Eslint plugin for format-js added.
```

### CODE REVIEW

**Review by m1lt0n — APPROVED** [review]

**Review by cwarnermm — APPROVED** [review]

**Review by hmhealey — CHANGES_REQUESTED** [review]
Shouldn't these changes to the default messages for some of these also include a change to `i18n/en.json` as well?

**Comment by cwarnermm** [comment]
Oh! Good catch, @hmhealey! Yes. @M-ZubairAhmed - any strings changed in code should also be updated in the ``en.json`` file as well. I believe there's a command you can run to update that file.

**Comment by M-ZubairAhmed** [comment]
oh i missed that. part of that i missed was because the CI didnt complain, then when i ran the command to extract the translations, nothing changed. It seem the tool is not catching change in white spaces. 
Anyways i updated by removing spaces manually.

File: components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx [hunk 1]
```diff
@@ -200,6 +200,7 @@ export default class SystemUsersDropdown extends React.PureComponent<Props, Stat
                                 <li>
                                     <FormattedMessage
                                         id='deactivate_member_modal.desc.for_users_with_bot_accounts1'
+                                        // eslint-disable-next-line formatjs/no-multiple-whitespaces
                                         defaultMessage='This action deactivates {username}<p>'
```
Comment (M-ZubairAhmed) [inline components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx:204]:
```suggestion
                                        defaultMessage='This action deactivates <p>{username}</p>'
```
should this be?

Comment (hmhealey) [reply components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx:204]: Actually, it shouldn't even have a `p` tag. Looks like it was added by mistake in https://github.com/mattermost/mattermost-webapp/pull/10793
```suggestion
                                        defaultMessage='This action deactivates {username}'
```

**Review by hmhealey — APPROVED** [review]
I was surprised that CI wasn't complaining about those either. I guess those are strings that `i18n-extract` can't handle for whatever reason.

This looks good to me now minus except for that random `<p>`

File: components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply components/admin_console/system_users/system_users_dropdown/system_users_dropdown.tsx]: oh, yes removed.

**Pull request merged**